### PR TITLE
Task 11/implement end to end

### PR DESCRIPTION
**Pull request by Adouleur**
Implement End-to-End (E2E) Testing with Cypress + added missing tests

### CODE REVIEW

File: src/components/atoms/ButtonComponent/ButtonComponent.spec.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { mount } from '@vue/test-utils';
+import ButtonComponent from '@/components/atoms/ButtonComponent/ButtonComponent.vue';
+
+describe('ButtonComponent', () => {
+  it('renders a button with default props', () => {
+    const wrapper = mount(ButtonComponent);
```
Comment (KHarpushkina) [inline src/components/atoms/ButtonComponent/ButtonComponent.spec.js]: Could you also add snapshot check. For all component test files

File: src/components/molecules/InfoBoard/InfoBoard.spec.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      }
+    });
+
+    // Check if the tasks left count is rendered correctly
+    expect(wrapper.find('.tasks-left').text()).toBe('3 tasks left');
+  });
+
+  it('calls the setFilter method when a filter button is clicked', async () => {
+    const setFilter = vi.fn();
+    const wrapper = mount(InfoBoard, {
```
Comment (KHarpushkina) [inline src/components/molecules/InfoBoard/InfoBoard.spec.js]: You don't need to call a mount for each test, just create a const at the top of the file and then use it in all the 'it' blocks.
You can use setProps to change the properties of your component

**Review by KHarpushkina — CHANGES_REQUESTED** [review]

File: src/components/molecules/InfoBoard/InfoBoard.spec.js (hunk 2 above)
Comment (Adouleur) [reply src/components/molecules/InfoBoard/InfoBoard.spec.js]: Done, thank you!

File: src/components/atoms/ButtonComponent/ButtonComponent.spec.js (hunk 1 above)
Comment (Adouleur) [reply src/components/atoms/ButtonComponent/ButtonComponent.spec.js]: Done! Thank you!

**Pull request merged**